### ISSUE TITLE
[Feature Request] Parallel processing

### ISSUE THREAD

**Issue by panangam** [KraXen72/shira#10]
This is a really neat library! Any chance it would be possible to add parallel processing for multiple songs? Between downloading the files and looking up metadata, I'm sure different songs can be processed in parallel for a much faster experience.

**Comment by KraXen72** [comment]
it is possible, but out of scope for now as i do not have time for that. I am happy to accept pull requests implementing this, though.

**Comment by KraXen72** [comment]
i've been learning more about paralellized programs in general. this might be something i'm interested in tackling, after all. No promises on an ETA however, as I'm busy with uni.
